File: Java/Nonmobile/Abstract_Classes/Figura.java
package com.example.javalib;

public abstract class Figura {

    double a; // musi być dziedziczone
    public Figura(double a){
        this.a = a;
    }
    public void oFigurze()
    {
        System.out.println("Jestem figurą");
        System.out.println("Pole wynosi: " + Pole());
        System.out.println("Obwód wynosi: " + Obwod());
    }

    public abstract double Pole();
    public abstract double Obwod();
}

File: C#/Abstract_Classes/KlasyAbstrakcyjne/Figura.cs
namespace KlasyAbstrakcyjne;

public abstract class Figura
{
    protected double a;

    public Figura(double a)
    {
        this.a = a;
    }
    public virtual void oFigurze()
    {
        Console.WriteLine("Jestem figurą");
        Console.WriteLine("Pole wynosi: " + Pole());
        Console.WriteLine("Obwód wynosi: " + Obwod());
    }
    public abstract double Pole();
    public abstract double Obwod();
}